numba/np/arrayobj.py

```diff
@@ -6102,6 +6102,48 @@ def impl(ary, indices_or_sections, axis=0):
         return np_array_split(ary, indices_or_sections, axis=axis)
 
 
+@overload(np.vsplit)
+def numpy_vsplit(ary, indices_or_sections):
+    if not isinstance(ary, types.Array):
+        msg = 'The argument "ary" must be an array'
+        raise errors.TypingError(msg)
+
+    if not isinstance(indices_or_sections, (types.Integer, types.Array)):
+        msg = ('The argument "indices_or_sections" '
+               'must be an integer or an array')
+        raise errors.TypingError(msg)
+
+    def impl(ary, indices_or_sections):
+        if ary.ndim < 2:
+            raise ValueError(('vsplit only works on '
+                              'arrays of 2 or more dimensions'))
+        return np.split(ary, indices_or_sections, axis=0)
+
+    return impl
+
+
+@overload(np.hsplit)
+def numpy_hsplit(ary, indices_or_sections):
+    if not isinstance(ary, types.Array):
+        msg = 'The argument "ary" must be an array'
+        raise errors.TypingError(msg)
+
+    if not isinstance(indices_or_sections, (types.Integer, types.Array)):
+        msg = ('The argument "indices_or_sections" '
+               'must be an integer or an array')
+        raise errors.TypingError(msg)
```

Comment:
Although NumPy says the second argument is an array, the following is also valid:

```python
import numpy as np
from numba import njit

@njit
def foo(a):
    return np.hsplit(a, [3, 6])

x = np.arange(16).reshape(4, 4)
# print(foo(x))
print(foo.py_func(x))
```